### PR TITLE
Remove support for surf HTTP client

### PR DESCRIPTION
**Pull request by djc**
## Changes

In a similar vein as #1534 / #1524, surf has seen its last release over 2 years ago and by the [admission](https://github.com/http-rs/surf/issues/352#issuecomment-1709404352) of its maintainer is pretty much unmaintained. On crates.io, it got 166k downloads in the last three months, vs reqwest's 11.7M.

#1534

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-http/CHANGELOG.md
Comment (hdost) [inline opentelemetry-http/CHANGELOG.md:7]: Add a **Breaking**

**Review by hdost — COMMENTED** [review]
Sounds good! Small ask on the Changelog, call it out as breaking.

**Review by shaun-cox — APPROVED** [review]
Thanks for the cleanup!

**Comment by cijothomas** [comment]
Also towards https://github.com/open-telemetry/opentelemetry-rust/issues/1427#issuecomment-1854700104

**Review by cijothomas — APPROVED** [review]

**Comment by cijothomas** [comment]
Good to merge once merge conflicts resolved, and the changelog calls out *breaking change*! Thanks @djc

**Comment by djc** [comment]
Rebased. How do you all feel about pushing out a release once this is merged? IMO it would be quite helpful to get the bumped tonic/prost dependencies out there given that they were severely lagging at this point.

**Review by lalitb — APPROVED** [review]
Thanks for all these cleanups

**Comment by cijothomas** [comment]
> Rebased. How do you all feel about pushing out a release once this is merged? IMO it would be quite helpful to get the bumped tonic/prost dependencies out there given that they were severely lagging at this point.

Agree. 
@jtescher @TommyCpp Can you help do a release with current state?

**Pull request merged**
